client/my-sites/site-settings/form-jetpack-monitor.jsx

```diff
@@ -1,212 +1,199 @@
+import { errorNotice, successNotice } from 'state/notices/actions';
+import {
+	getSiteMonitorSettings,
+	isActivatingJetpackModule,
+	isDeactivatingJetpackModule,
+	isFetchingJetpackModules,
+	isJetpackModuleActive,
+	isMonitorSettingsUpdateSuccessful,
+	isRequestingSiteMonitorSettings,
+	isUpdatingSiteMonitorSettings
+} from 'state/selectors';
+
+class SiteSettingsFormJetpackMonitor extends Component {
+	state = {};
+
+	componentWillReceiveProps( nextProps ) {
+		if ( isEmpty( this.state ) && nextProps.monitorSettings ) {
+			this.setState( nextProps.monitorSettings );
+		}
+	}
 
-	getModuleStatus: function( site ) {
-		site = site || this.props.site;
-
-		site.fetchMonitorSettings( ( function( error, data ) {
-			if ( ! error ) {
-				this.setState( {
-					emailNotifications: data.settings.email_notifications,
-					wpNoteNotifications: data.settings.wp_note_notifications
-				} );
-			} else {
-				debug( 'error getting Monitor settings', error );
-			}
-		} ).bind( this ) );
+	recordEvent = ( event ) => {
+		return () => {
+			this.props.trackEvent( event );
+		};
+	}
 
+	handleToggle = name => () => {
+		this.props.trackEvent( `Toggled ${ name }` );
 		this.setState( {
-			enabled: site.isModuleActive( 'monitor' )
+			...this.state,
+			[ name ]: ! this.state[ name ],
 		} );
-	},
+		this.saveSettings();
+	}
 
-	prompt: function() {
-		return (
-			<div className="site-settings__jetpack-prompt">
-				<img src="/calypso/images/jetpack/illustration-jetpack-monitor.svg" width="128" height="128" />
-
-				<div className="site-settings__jetpack-prompt-text">
-					<p>{ this.translate( "Automatically monitor your website and make sure it's online." ) }</p>
-					<p>
-						{ this.translate(
-							"We'll periodically check your site from our global network of servers to make sure it's online, and email you if it looks like your site is not responding for any reason."
-						) }
-					</p>
-				</div>
+	saveSettings = () => {
+		const { monitorSettingsUpdateSuccessful, siteId, translate } = this.props;
 
-			</div>
-		);
-	},
-
-	saveSettings: function() {
-		this.setState( { submitingForm: true } );
-		notices.clearNotices( 'notices' );
-		this.props.site.updateMonitorSettings( this.state.emailNotifications, this.state.wpNoteNotifications, ( function( error ) {
-			if ( error ) {
-				this.handleError();
-				notices.error( this.translate( 'There was a problem saving your changes. Please, try again.' ) );
+		this.props.updateSiteMonitorSettings( siteId, this.state ).then( () => {
+			if ( ! monitorSettingsUpdateSuccessful ) {
+				this.props.errorNotice( translate( 'There was a problem saving your changes. Please, try again.' ) );
 				return;
 			}
-			notices.success( this.translate( 'Settings saved successfully!' ) );
-			this.props.markSaved();
-			this.setState( { submittingForm: false } );
-		} ).bind( this ) );
-	},
+			this.props.successNotice( translate( 'Settings saved successfully!' ) );
```

Comment:
Since we're not setting any component state, we might be able to move this to the middleware handler for `SITE_MONITOR_SETTINGS_UPDATE_SUCCESS` and `SITE_MONITOR_SETTINGS_UPDATE_FAILURE`. 

If we choose not to, I'd recommend rewriting this part in `componentWillReceiveProps` in a future PR since we're moving away from thunks.